metadata-ingestion/src/datahub/ingestion/source/redash.py

```diff
@@ -0,0 +1,462 @@
+        https://redash.io/help/user-guide/visualizations/visualization-types
+        Redash has multiple visualization types. Chart type is actually Plotly.
+        So we need to check options returned by API, which series type is being used.
+        """
+        viz_type = viz_data.get("type", "")
+        viz_options = viz_data.get("options", {})
+        globalSeriesType = viz_options.get("globalSeriesType", "")
+        report_key = f"redash-chart-{viz_data['id']}"
+
+        # handle Plotly chart types
+        if viz_type == "CHART":
+            chart_type = PLOTLY_CHART_MAP.get(
+                globalSeriesType, DEFAULT_VISUALIZATION_TYPE
+            )
+            if not chart_type:
+                self.report.report_warning(
+                    key=report_key,
+                    reason=f"ChartTypeClass={viz_data['type']} with options.globalSeriesType={globalSeriesType} is missing. Setting to None",
+                )
+
+        chart_type = VISUALIZATION_TYPE_MAP.get(viz_type, DEFAULT_VISUALIZATION_TYPE)
+        if not chart_type:
+            self.report.report_warning(
+                key=report_key,
+                reason=f"ChartTypeClass={viz_data['type']} is missing. Setting to None",
+            )
+
+        return chart_type
+
+    def _get_chart_snapshot(self, query_data, viz_data):
+        chart_urn = f"urn:li:chart:({self.platform},{viz_data['id']})"
+        chart_snapshot = ChartSnapshot(
+            urn=chart_urn,
+            aspects=[],
+        )
+
+        modified_actor = f"urn:li:corpuser:{(viz_data.get('changed_by') or {}).get('username', 'unknown')}"
+        modified_ts = int(
+            dp.parse(viz_data.get("updated_at", "now")).timestamp() * 1000
+        )
+        title = f"{query_data.get('name')} {viz_data.get('name', '')}"
+
+        last_modified = ChangeAuditStamps(
+            created=AuditStamp(time=modified_ts, actor=modified_actor),
+            lastModified=AuditStamp(time=modified_ts, actor=modified_actor),
+        )
+
+        # Getting chart type
+        chart_type = self._get_chart_type_from_viz_data(viz_data)
+        chart_url = f"{self.config.connect_uri}/queries/{query_data.get('id')}#{viz_data.get('id', '')}"
+        description = viz_data.get("description") if viz_data.get("description") else ""
+
+        # TODO: Getting table lineage from SQL parsing
+        # Currently we only get database level source from `data_source_id` which returns database name or Bigquery's projectId
+        # query = query_data.get("query", "")
+
+        data_source_id = query_data.get("data_source_id")
+        datasource_urns = [
+            self._get_datasource_urn_from_data_source_id(data_source_id),
+        ]
+
+        chart_info = ChartInfoClass(
+            type=chart_type,
+            description=description,
+            title=title,
+            lastModified=last_modified,
+            chartUrl=chart_url,
+            inputs=datasource_urns if datasource_urns else None,
```

Comment:
@datascienceChris and this alse may be the culprit of why the chart UI showing error